pallets/ajuna-affiliate/src/tests.rs

```diff
@@ -0,0 +1,251 @@
+use crate::{mock::*, *};
+use frame_support::{assert_noop, assert_ok};
+use sp_runtime::bounded_vec;
+
+mod add_rule {
+	use super::*;
+
+	#[test]
+	fn add_rule_should_work() {
+		ExtBuilder::default().build().execute_with(|| {
+			let extrinsic_id = (0, 0);
+			let rule = 1;
+
+			assert_ok!(<Affiliates as RuleMutator<AccountIdOf<Test>>>::try_add_rule_for(
+				extrinsic_id,
+				rule
+			));
+
+			System::assert_last_event(mock::RuntimeEvent::Affiliates(crate::Event::RuleAdded {
+				extrinsic_id,
+			}));
+
+			assert_eq!(AffiliateRules::<Test>::get(extrinsic_id), Some(rule));
+		})
+	}
+
+	#[test]
+	fn cannot_add_rule_to_already_marked_extrinsic() {
+		ExtBuilder::default().build().execute_with(|| {
+			let extrinsic_id = (0, 0);
+			let rule = 1;
+			assert_ok!(<Affiliates as RuleMutator<AccountIdOf<Test>>>::try_add_rule_for(
+				extrinsic_id,
+				rule
+			));
+
+			assert_noop!(
+				<Affiliates as RuleMutator<AccountIdOf<Test>>>::try_add_rule_for(
+					extrinsic_id,
+					rule
+				),
+				Error::<Test>::ExtrinsicAlreadyHasRule
+			);
+		})
+	}
+}
+
+mod clear_rule {
+	use super::*;
```

Comment:
I think we miss the test of removing a nonexistent rule.

Reply:
Hmm, well nothing happens if you do that in the current logic, so I didn't deem a test necessary